### ISSUE TITLE
`hugo serve` executes "partials/footer/script-footer.html" failed

### ISSUE THREAD

**Issue by hitzhangjie** [h-enk/doks#572]
## Description

Hi, when I run `hugo serve`, it reports some error like:

```
blog/single.html:28:7: executing "blog/single.html" at <partial "footer/script-footer.html" .>: error calling partial: "path-to/layouts/partials/footer/script-footer.html:37:23": execute of template failed: template: partials/footer/script-footer.html:37:23: executing "partials/footer/script-footer.html" at <append $flexSearch>: error calling append: reflect: call of reflect.Value.Type on zero Value.
```

I don't know why. What's interesting is it works well on my macbook pro (intel chip + macOS catalina + go1.16 + hugo0.91.0-extended), but failed on my new macbook pro (m1 chip + macOS monterey + go1.17 + hugo0.91.0-extended).

I changed go version from go1.17 to go1.16, it still reports the same problem.

### Steps to reproduce

This is my blog repo: `https://github.com/hitzhangjie/myspace`, run `hugo serve` in the repo directory.

### Expected result

`hugo serve` runs normally.

### Actual result

`hugo serve` reports error mentioned above.

### Environment

```bash
npm run check

> @hitzhangjie/myspace@0.2.3 precheck
> npm version

{
  '@hitzhangjie/myspace': '0.2.3',
  npm: '8.1.4',
  node: '17.2.0',
  v8: '9.6.180.14-node.12',
  uv: '1.42.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.18.1',
  modules: '102',
  nghttp2: '1.46.0',
  napi: '8',
  llhttp: '6.0.4',
  openssl: '1.1.1l',
  cldr: '39.0',
  icu: '69.1',
  tz: '2021a',
  unicode: '13.0'
}

> @hitzhangjie/myspace@0.2.3 check
> hugo version

hugo v0.91.0-DEV+extended darwin/arm64 BuildDate=unknown
```

**Comment by h-enk** [comment]
Hi, you're running an older version of Doks with newer dependencies, that's why you're seeing this error.

Replace `./layouts/partials/footer/script-footer.html` line 37:

```html
{{ $flexSearch := resources.Get "js/vendor/flexsearch/dist/flexsearch.min.js" -}}
```

with: 

```html
{{ $flexSearch := resources.Get "js/vendor/flexsearch/dist/flexsearch.bundle.js" -}}
```

**Comment by h-enk** [comment]
@hitzhangjie, good to hear!

The file does exist — virtually that it is. It is mounted in your [config.toml](https://github.com/hitzhangjie/myspace/blob/master/config/_default/config.toml#L93-L95) and appended to `main.js` in your [script-footer.html](https://github.com/hitzhangjie/myspace/blob/master/layouts/partials/footer/script-footer.html#L35-L38)

The Hugo mount functionality is very handy + powerful. For more background, see also the Hugo docs: [Module Config: mounts](https://gohugo.io/hugo-modules/configuration/#module-config-mounts)

**Comment by h-enk** [comment]
BTW, I noted that your search is broken:

![Snag_18d79095](https://user-images.githubusercontent.com/3902872/175060402-ac705da1-008c-449d-baf5-b95d869b6a15.png)

I think you should (at least) update your line 67 of [index.js](https://github.com/hitzhangjie/myspace/blob/master/assets/js/index.js#L67) for line 66  of Doks' [index.js](https://github.com/h-enk/doks/blob/master/assets/js/index.js#L66). Maybe more updates are needed.

Could you please report back when you got it working again? I'm asking, for I'm interested in how it works for you with Chinese tokens. This could be valuable input for Doks multilingual improvements. Thanks!

**Comment by hitzhangjie** [comment]
@h-enk Thank you very much to show me the error. I do as following:
- new FlexSearch changed to new FlexSearch.Document
- index.add(docs) changed to index.add({...})
- ...

After that, it reports no error, but index.search returns empty, I think it maybe relevant to the 'tokenize' setting, but I used 'forward' and the setting before, it still returns empty. I think I messed up the the dependencies :)
So, I just update the dokz theme to the latest by clone my-dokz-site, and copy my content there. Then I do a little adjustment to the layouts, it worked now.

I have several suggestions, I will submit new issue or PR to discuss these with you.
